apps/prairielearn/src/lib/workspace.js

```diff
@@ -451,6 +496,39 @@ module.exports = {
       }
     }
 
+    if (fileGenerationErrors.length > 0) {
+      const output = fileGenerationErrors.map((error) => `${error.file}: ${error.msg}`).join('\n');
+      issues.insertIssue({
+        variantId: variant.id,
+        studentMessage: 'Error retrieving workspace files',
+        instructorMessage: 'Error retrieving workspace files',
+        manuallyReported: false,
+        courseCaused: true,
+        courseData: { workspace, variant, question, course },
+        systemData: {
+          courseErrData: {
+            // This is shown in the console log of the issue
+            outputBoth: output,
+            // This data is only shown if user is admin (e.g., in dev mode).
+            errors: fileGenerationErrors.map((error) => ({
+              ...error,
+              // Since error is typically not serializable, a custom object is created.
```

Comment:
In case you don't mind another dependency, I've used `serialize-error` elsewhere in the PL codebase:

https://github.com/PrairieLearn/PrairieLearn/blob/6d28c3f544e7874d4ed2167204ef7ab0b53f0e5e/packages/migrations/src/batched-migrations/batched-migration-runner.ts#L186

Reply:
Done in [441eb4e](https://github.com/PrairieLearn/PrairieLearn/pull/8686/commits/441eb4e737b2e1e08ead44a10200bdd6bb048566).